src/phases/command-phase.ts

```diff
@@ -179,7 +180,8 @@ export class CommandPhase extends FieldPhase {
     case Command.POKEMON:
     case Command.RUN:
       const isSwitch = command === Command.POKEMON;
-      if (!isSwitch && this.scene.arena.biomeType === Biome.END) {
+      const mysteryEncounterFleeAllowed = this.scene.currentBattle.mysteryEncounter?.fleeAllowed;
+      if (!isSwitch && (this.scene.arena.biomeType === Biome.END || (!isNullOrUndefined(mysteryEncounterFleeAllowed) && !mysteryEncounterFleeAllowed))) {
```

Comment:
NIT: potentially make use of destructuring:

```suggestion
      const { currentBattle, arena } = this.scene;
      const mysteryEncounterFleeAllowed = currentBattle.mysteryEncounter?.fleeAllowed;
      if (!isSwitch && (arena.biomeType === Biome.END || (!isNullOrUndefined(mysteryEncounterFleeAllowed) && !mysteryEncounterFleeAllowed))) {
```